### PR TITLE
ls: provide `-s` & `-k` option

### PR DESCRIPTION
**Pull request by aburn**
**work in progress** 

This is a preliminary set of changes for issue tracked here: https://github.com/uutils/coreutils/issues/2257 

Currently this is based of `ls` utility on `OS X`. it was pointed out in the discussion on the issue page, that we need to align with GNU. I am working on those changes and will keep updating the PR.

That said, I wanted to get some early feedback regarding coding conventions & style used in the project and anything else, that would help me do the rest of the changes in according to the feedback provided.

Current status and sample outputs are captured in the comment : https://github.com/uutils/coreutils/issues/2257#issuecomment-890402288 

**Pending tasks**
- make the changes compatible to GNU.
- Add/update test cases to cover the new options (`-s` & `-k`).
- Current code has `#[cfg(unix)]` tagged in the code. It was mentioned in the discussion, that we can use the CI tests to test on `Windows` platforms as-well. I will get to this once the first task is close to completion.

### CODE REVIEW

File: src/uu/ls/src/ls.rs [hunk 1]
```diff
@@ -1765,6 +1840,18 @@ fn classify_file(path: &PathData) -> Option<char> {
...
+            name = path.md().map_or_else(
+                || "?".to_string(),
+                |md| (md.blocks() * 512 / block_size).to_string(),
+            ) + " "
+                + &name;
+        }
+    }
+
     #[cfg(unix)]
     {
```
Comment (miDeb) [inline src/uu/ls/src/ls.rs:1856]: You could combine these two blocks.

File: src/uu/ls/src/ls.rs [hunk 2]
```diff
@@ -1522,6 +1585,18 @@ fn display_item_long(
         }
     }
 
+    #[cfg(unix)]
+    {
+        if config.block_size.is_some() {
+            let block_size = config.block_size.map_or_else(get_env_block_size, |v| v);
+            let _ = write!(
```
Comment (miDeb) [inline src/uu/ls/src/ls.rs:1592]: You are ignoring the result of `write!` here. This is usually not the best solution. On the other hand the surrounding code does the same, so I think it's fine (maybe there's a reason behind it, but I don't know)

File: src/uu/ls/src/ls.rs [hunk 3]
```diff
@@ -1504,6 +1566,7 @@ fn display_item_long(
     item: &PathData,
     max_links: usize,
     max_size: usize,
+    _max_blk_width: usize,
```
Comment (miDeb) [inline src/uu/ls/src/ls.rs:1569]: Variables prefixed with an underscore are usually unused. It's better to add a `cfg` on the parameter and drop the underscore.

**Review by miDeb — COMMENTED** [review]
FYI: [The GNU manual](https://www.gnu.org/software/coreutils/manual/html_node/ls-invocation.html) should have most of the information you need. For example, the long forms of `-s` and `-k` are different from what you currently have.

If you go to the "files" tab on this pr you can see the compile errors that occur on windows.

Below are some minor comments on style issues. Let us know when this is ready for a full review.

File: src/uu/ls/src/ls.rs (hunk 1 above)
Comment (aburn) [reply src/uu/ls/src/ls.rs:1856]: Will do. Thanks!

**Comment by aburn** [comment]
@miDeb : Thanks for your comments. Will fix them and address in the next iter.

> FYI: [The GNU manual](https://www.gnu.org/software/coreutils/manual/html_node/ls-invocation.html) should have most of the information you need. For example, the long forms of `-s` and `-k` are different from what you currently have.

Working on making this GNU style. Thanks for the pointers to the documentation.

> If you go to the "files" tab on this pr you can see the compile errors that occur on windows.

First focussing on getting the functionality right on the Unix platforms. Will fix windows CI failures as part of the next iteration too.

> Below are some minor comments on style issues. Let us know when this is ready for a full review.
Will do. I will remove the Draft tag from this PR once ready for full review. Thanks a lot!

**Comment by sylvestre** [comment]
@aburn ping?

**Comment by sylvestre** [comment]
no activity for a while, closing
please reopen when ready :)

**Pull request closed without merging**